test/xaveFxPool.integration.spec.ts

```diff
@@ -0,0 +1,142 @@
+// yarn test:only test/xaveFxPool.integration.spec.ts
+import dotenv from 'dotenv';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { SOR, SubgraphPoolBase, SwapTypes } from '../src';
+import { ADDRESSES, Network, vaultAddr } from './testScripts/constants';
+import { parseFixed } from '@ethersproject/bignumber';
+import { expect } from 'chai';
+import { Vault__factory } from '@balancer-labs/typechain';
+import { AddressZero } from '@ethersproject/constants';
+import { setUp } from './testScripts/utils';
+
+/*
+ * Testing Notes:
+ * - Add polygon ALCHEMY_URL in .env
+ * - Change --fork-block-number to 38546978
+ * - Change hardhat.config.ts chainId to 137
+ * - Run polygon node on terminal: yarn run node
+ */
+
+dotenv.config();
+
+let sor: SOR;
+const networkId = Network.POLYGON;
```

Comment:
I noticed you have some mainnet pools. Would it be possible to update this to use mainnet, (this already exists in our CI)?

Reply:
we dont have much there yet. most of our liquidity is in polygon atm. what would be the effect if we dont?

Reply:
added a mainnet implementation already :)